gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageObjectAttributes.java

```diff
@@ -131,25 +145,27 @@ public FileTime lastAccessTime() {
 
   @Override
   public Object fileKey() {
-    return metadata.getFilename();
+    return name;
```

Comment:
Would it be possible for us to include the bucket, name, *and* generation? This is a truly unique canonical name that allows equality between two file keys to guarantee the contents are the same. That is an advantageous behavior.